composer/utils/iter_helpers.py

```diff
@@ -88,3 +90,37 @@ def iterate_with_pbar(iterator, progress_bar=None):
             yield x
             if pb is not None:
                 pb.update(len(x))
+
+
+def iter_to_stream(iterator: Iterator[bytes], buffer_size: int = io.DEFAULT_BUFFER_SIZE) -> io.BufferedReader:
+    """Function to convert iterator of bytes into a file-like binary stream object.
+
+    Original implementation found `here <https://stackoverflow.com/questions/6657820/how-to-convert-an-iterable-to-a-stream/20260030#20260030>`_.
+    
+    Args:
+        iterator (Iterator[bytes]): An iterator over bytes objects
+        buffer_size (int): Buffer length of the stream
+    
+    Returns:
+        io.BufferedReader: A buffered binary stream.
+    """
+
+    class BytesToStream(io.RawIOBase):
```

Comment:
Can this class be defined outside of the function, and then take `iterator` as part of the init signature? In addition, could we call the class `IteratorFileStream`? Then, I think we can remove the function `iter_to_stream`. The caller can then do `f = IteratorFileStream(iterator)`, or if they need a buffered reader, `f = io.BufferedReader(IteratorFileStream(iterator), buffer_size=buffer_size)`